### ISSUE TITLE
Fix and document proxy configuration

### ISSUE THREAD

**Issue by RPCMoritz** [andypetrella/spark-notebook#184]
Current application.conf proxy configuration isn't well documented (repo vs. http, which format is being used?), and probably slightly counterproductive, whe usually proxy settings are available in system properties (somewhat OS-independent, http_proxy etc.)
The current settings also do not appear to work using a naive configuration.

**Comment by andypetrella** [comment]
Agreed, could you give a small example of non working case (using naive conf f.i.)?

So it'll be easier to address

**Comment by RPCMoritz** [comment]
The following settings 
```
http {
  proxyHost="localhost"
  proxyPort=8383
  #proxyUser=...
  #proxyPassword=...
}

remote-repos {
  proxy {
    "protocol" = https,
    "host"     = "localhost",
    "port"     = 8383,
    # "username" = ...,
    # "password" = ...
  }
}
```
are expected to match with the following env:
```
http_proxy=http://localhost:8383
https_proxy=http://localhost:8383
```
which is respected by maven and and wget, for example.

**Comment by RPCMoritz** [comment]
Since we gittered about getting a reproducable config, and you're a docker-fan:
I found this docker-image: https://github.com/Painted-Fox/docker-tinyproxy which could be used as the proxy, with the following ENV passed to a modified notebook container:
```
ENV http_proxy="http://192.168.1.1:3128"
ENV https_proxy="https://192.168.1.1:3128"
```
The modification relates to the network setup, you want to block direct network access
and setup a direct connection between the proxy-container and the notebook container, as described here:
https://docs.docker.com/articles/networking/#container-networking (and below, particularly for point-to-point)
With some trick routing (default route over the p2p interface) you should be able to both connect to the notebook and prevent it from accessing the 'net without going through the proxy. At worst, use iptables to reject all packets on the docker bridge originating from the eth0-ip of the proxied-notebook container

It's a bit of a mess, but probably easier than setting up a more complex SSH-proxy in docker. 
With other-than-docker containers this may still be the easier way.
